stdlib/2/urllib2.pyi

```diff
@@ -128,19 +128,23 @@ class ProxyDigestAuthHandler(BaseHandler, AbstractDigestAuthHandler):
     handler_order = ...  # type: int
     def http_error_407(self, req, fp, code, msg, headers): ...
 
-class AbstractHTTPHandler(BaseHandler):
+class AbstractHTTPHandler(BaseHandler):  # undocumented
     def __init__(self, debuglevel: int = ...) -> None: ...
-    def do_request_(self, request): ...
-    def do_open(self, http_class, req): ...  # undocumented
+    def set_http_debuglevel(self, level: int) -> None: ...
+    def do_request_(self, request: Request) -> Request: ...
+    def do_open(self,
+                http_class: HTTPConnectionProtocol,
+                req: Request,
+                **http_conn_args: Optional[ssl.SSLContext]) -> HTTPResponse: ...
```

Comment:
The implementation seems to return an `addinfourl` instance. And `http_conn_args` should probably be `Any` (see above).